src/modules/map/components/cesium/CesiumGPXLayer.vue

```diff
@@ -0,0 +1,159 @@
+    log.debug('Remove GPX layer')
+    if (gpxDataSource.value) {
+        getViewer().dataSources.remove(gpxDataSource.value)
+        gpxDataSource.value = null
+        getViewer().scene.requestRender() // Request a render after removing the DataSource
+    }
+    isPresentOnMap.value = false
+}
+
+function updateStyle() {
+    // Function to create a red circle image using a canvas
+    function createRedCircleImage(radius) {
+        // Create a new canvas element
+        const canvas = document.createElement('canvas')
+        const context = canvas.getContext('2d')
+
+        // Set the canvas sizes
+        canvas.width = radius * 2
+        canvas.height = radius * 2
+
+        // Draw a red circle on the canvas
+        context.beginPath()
+        context.arc(radius, radius, radius, 0, 2 * Math.PI, false)
+
+        const color = `rgba(255, 0, 0, ${opacity.value})`
+        context.fillStyle = color
+        context.fill()
+
+        // Return the data URL of the canvas drawing
+        return canvas.toDataURL()
+    }
+
+    // Create a red circle image with a radius of 8 pixels
+    const radius = 8
+    const billboardSize = radius * 2
+    const redCircleImage = createRedCircleImage(radius)
+
+    const redCircleBillboard = new BillboardGraphics({
+        image: redCircleImage,
+        width: billboardSize,
+        height: billboardSize,
+        eyeOffset: new Cartesian3(0, 0, -100), // Make sure the billboard is always seen
+        heightReference: HeightReference.CLAMP_TO_TERRAIN, // Make the billboard always appear on top of the terrain
+    })
+    const redColorMaterial = new ColorMaterialProperty(Color.RED.withAlpha(opacity.value))
+
+    const entities = gpxDataSource.value._entityCollection.values
+
+    for (let i = 0; i < entities.length; i++) {
+        const entity = entities[i]
+        if (opacity.value === 0) {
+            entity.show = false
+            continue
+        }
+        // Hide the billboard for billboard on the lines by checking if there is a description
+        // Imported GPX files from web-mapviewer have a description for the waypoints
+        // This might be not working for generic GPX files
+        if (cesiumDefined(entity.billboard)) {
+            if (!entity.description) {
+                entity.show = false
+            } else {
+                entity.show = true
+                entity.billboard = redCircleBillboard
+            }
+        }
+
+        if (cesiumDefined(entity.polyline)) {
+            entity.show = true
+            entity.polyline.material = redColorMaterial
+            entity.polyline.width = 1.5
+        }
+
+        if (cesiumDefined(entity.polygon)) {
+            entity.show = true
+            entity.polygon.material = redColorMaterial
+            entity.polygon.outline = true
+            entity.polygon.outlineColor = Color.BLACK
+        }
+    }
+    getViewer().scene.requestRender()
+}
+</script>
+<template>
+    <div>
+        <slot />
+    </div>
```

Comment:
no need to wrap in a div, simply give the `<slot />` as template, and no HTML will be generated (no empty div) when the component is loaded